### ISSUE TITLE
Markdown files defining their own layout do not get rendered within that layout

### ISSUE THREAD

**Issue by carbontwelve** [carbontwelve/tapestry#56]
Steps to reproduce:

Create the file `source\test.md` with the following content:

```
---
layout: page
title: "Test"
---

Test Content
```

Create the file `source\_templates\page.phtml` with the following content:

```
<h1><?= (isset($title) ? $title : ''); ?></h1>
<div><?= $this->section('content') ?></div>
```

Expected outcome is that upon building you end up with the file `test.html` with the following content:

```
<h1>Test</h1>
<div><p>Test Content</p></div>
```

What you actually get is `<div><p>Test Content</p></div>` the markdown is rendered out to `.html` and never gets passed through to a `phtml` template, defined or otherwise.

**Comment by carbontwelve** [comment]
This is dealt with by bf1c70dfd332659aebacd71d9a716de4b90ebaf2 however a markdown file that uses a child template does not work as within `Tapestry\Plates\Template` the `$file` variable never gets set to the `File` object for that templates linked file.

Because this is a side effect of this issue I am continuing development of that bug within this issue number.

**Comment by carbontwelve** [comment]
When a template extends another template the `File` class was not being passed along which meant that the view helper was receiving `null` when trying to get the files rendered content to display.

This is now completely fixed by 99209964c044611c71c2ad335a4a0ff5aa7a6433 and will be pulled into development branch.
